src/feature-flags.ts

```diff
@@ -17,11 +17,13 @@ const MINIMUM_ENABLED_CODEQL_VERSION = "2.11.6";
 export type CodeQLDefaultVersionInfo =
   | {
       cliVersion: string;
+      toolsFeatureFlagsValid?: boolean;
```

Comment:
We know we're on Dotcom here, so `toolsFeatureFlagsValid` should always be present

```suggestion
      toolsFeatureFlagsValid: boolean;
```

Reply:
This ended up being somewhat difficult to implement without adding some ugly casts, so I've left it optional.